components/lvgl/widgets.rst

```diff
@@ -0,0 +1,1790 @@
+- **width** (*Optional*): Width of the widget in pixels or a percentage, or ``SIZE_CONTENT``.
+
+.. note::
+
+    The size settings support a special value: ``SIZE_CONTENT``. It means the widget's size in the respective direction will be set to the size of its children. Note that only children on the right and bottom sides will be considered and children on the top and left remain cropped. This limitation makes the behavior more predictable. Widgets with ``hidden`` or ``floating`` flags will be ignored by the ``SIZE_CONTENT`` calculation.
+    
+    Similarly to CSS, LVGL also supports ``min_width``, ``max_width``, ``min_height`` and ``max_height``. These are limits preventing a widget's size from becoming smaller/larger than these values. They are especially useful if the size is set by percentage or ``SIZE_CONTENT``.
+
+- **min_width**, **max_width**, **min_height**, **max_height** (*Optional*, int16 or percentage): Sets a minimal/maximal width or a minimal/maximal height. Pixel and percentage values can be used. Percentage values are relative to the dimensions of the parent's content area. Defaults to ``0%``.
+- **scrollbar_mode** (*Optional*, string): If a child widget is outside its parent content area (the size without padding), the parent can become scrollable (see the ``scrollable`` :ref:`flag <lvgl-widget-flags>`). The widget can either be scrolled horizontally or vertically in one stroke. Scroll bars can appear depending on the setting:
+    - ``"OFF"``: Never show the scroll bars (use the double quotes!).
+    - ``"ON"``: Always show the scroll bars (use the double quotes!).
+    - ``"ACTIVE"``: Show scroll bars while a widget is being scrolled.
+    - ``"AUTO"``: Show scroll bars when the content is large enough to be scrolled (default).
+
+- **align** (*Optional*, dict): Alignment of the of the widget relative to the parent. A child widget is clipped to its parent boundaries. One of the values *not* starting with ``OUT_`` (see picture below).
+- **align_to** (*Optional*, list): Alignment of the of the widget relative to another widget on the same level:
+    - **id** (**Required**): The ID of a widget *to* which you want to align.
+    - **align** (**Required**, string): Desired alignment (one of the values starting with ``OUT_``).
+    - **x** (*Optional*, int16 or percentage): Horizontal offset position. Default ``0``.
+    - **y** (*Optional*, int16 or percentage): Vertical offset position. Default ``0``.
+
+.. figure:: /components/lvgl/images/lvgl_align.png
+    :align: center
+
+- **group** (*Optional*, string): The name of the group of widgets which will interact with a  :doc:`/components/sensor/rotary_encoder`. In every group there is always one focused widget which receives the encoder actions. You need to associate an input device with a group. An input device can send key events to only one group but a group can receive data from more than one input device.
```

Comment:
```suggestion
- **group** (*Optional*, string): The name of the group of widgets which will interact with a  :doc:`/components/sensor/rotary_encoder`. In every group there is always one focused widget which receives the encoder actions. You need to associate an input device with a group. An input device can send key events to only one group but a group can receive data from more than one input device. If no group is specified for a widget or an encoder, an unnamed default group will be assigned, so in most cases where only one encoder is used it will not be necessary to explicitly specify a group.
```